pkg/agent/plugin/workloadattestor/windows/windows_windows.go

```diff
@@ -0,0 +1,200 @@
+//go:build windows
+// +build windows
+
+package windows
+
+import (
+	"context"
+	"fmt"
+	"syscall"
+
+	"github.com/hashicorp/go-hclog"
+	workloadattestorv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/agent/workloadattestor/v1"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/pkg/common/telemetry"
+	"golang.org/x/sys/windows"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func builtin(p *Plugin) catalog.BuiltIn {
+	return catalog.MakeBuiltIn(pluginName, workloadattestorv1.WorkloadAttestorPluginServer(p))
+}
+
+func New() *Plugin {
+	p := &Plugin{q: &processQuery{}}
+	return p
+}
+
+type Plugin struct {
+	workloadattestorv1.UnsafeWorkloadAttestorServer
+
+	log hclog.Logger
+	q   processQueryer
+}
+
+type processInfo struct {
+	pid        int32
+	user       string
+	userSID    string
+	groups     []string
+	groupsSIDs []string
+}
+
+func (p *Plugin) Attest(ctx context.Context, req *workloadattestorv1.AttestRequest) (*workloadattestorv1.AttestResponse, error) {
+	process, err := p.newProcessInfo(req.Pid, p.q)
```

Comment:
nit: don't pass p.q since newProcessInfo can already get to it through `p` (passing it brings up questions like "when would i want to pass something else")

Reply:
Thanks for noticing this!